src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMParams.java

```diff
@@ -0,0 +1,1179 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.broadinstitute.barclay.argparser.Advanced;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+
+/**
+ * This class represents the hyper-parameters and parameters for {@link CoverageModelEMAlgorithm}
+ *
+ * Works both as an argument collection as well as a getter/setter.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class CoverageModelEMParams {
+
+    public enum PsiUpdateMode {
+        /**
+         * Target-resolved unexplained variance
+         */
+        PSI_TARGET_RESOLVED,
+
+        /**
+         * Isotropic unexplained variance (i.e. one value for all targets)
```

Comment:
Isotropic refers to the covariance matrix of the noise in PPCA/FA, which is isotropic in PPCA and target-dependent in FA. I don't have a strong opinion here; would you prefer `PSI_HOMOGENEOUS` and `PSI_TARGET_DEPENDENT` instead?